Shell/FunctionDefinition.cpp

```diff
@@ -723,13 +722,8 @@ Clause* FunctionDefinition::applyDefinitions(Clause* cl)
     UnitList::push(defCl, premises);
   }
   UnitList::push(cl, premises);
-  Clause* res = new(clen) Clause(clen, NonspecificInferenceMany(InferenceRule::DEFINITION_UNFOLDING, premises));
-  res->setAge(cl->age());
-
-  for(unsigned i=0;i<clen;i++) {
-    (*res)[i] = resLits[i];
-  }
-
+  auto res = Clause::fromStack(*resLits, NonspecificInferenceMany(InferenceRule::DEFINITION_UNFOLDING, premises));
+  res->setAge(cl->age()); // TODO isn't this dones automatically?
```

Comment:
So here, the convention is (as with simplifications) that the age is inherited from the first premise...